### DISCUSSION TITLE
How does the blog generate the filter pages?

### DISCUSSION THREAD

**Discussion by PetersOtto** [datenstrom/community#933] [Ask a question] [answered]
Hello everyone!

I'm stuck and need your help.
I have set up a website with a blog and a news page. Just as I have described here: https://github.com/datenstrom/yellow/discussions/849#discussioncomment-7726671

Unfortunately I can't find a way to generate the »filter pages«. For example `www.your-domain/news/tag:yourtag/`
Does anyone have any ideas? How are the »filter pages« generated by Yellow?

&#128587;

**Comment by PetersOtto** [comment]
My findings so far are that you have to change the `blog.php` to be able to use filter pages. My solution here https://github.com/datenstrom/yellow/discussions/849#discussioncomment-7726671 does not work with filtering. 

The following way works, but requires more work when the blog extension will be updated

1. download the blog extension https://github.com/annaesvensson/yellow-blog
2. unpack the block extension
3. rename the folder from `yellow-blog-main` to `yellow-news-main`
4. change into the folder `yellow-news-main`
5. open a terminal in this folder (`yellow-news-main`)
5. rename into the files all blogs to news (`find  -type f -exec sed -i 's/blog/news/g' {} \;`)
6. rename into the files all Blogs to News (`find  -type f -exec sed -i 's/Blog/News/g' {} \;`)
7. rename the parts blog to news in the filename (`rename 's/blog/news/' *`)
8. check if everything has been renamed correctly
9. compress the folder `yellow-news-main` to `yellow-news-main.zip`
10. copy the file `yellow-news-main.zip` into the extensions folder of your Yellow installation
11. go to the `content` folder and change the folder name from `2-news` to `3-news`. Or choose a number that matches better to your website.
12. Open the `yellow-language.ini` (in my case german) in the extension folder and add the following lines: 
```
NewsDescription: Blog für deine Webseite.
NewsBy: von
NewsTag: Tags:
NewsMore: Lies mehr…
```
Now the second blog should work.
**Be careful not to rename the wrong files!** I work with Ubuntu 20.04.

**Comment by markseuffert** [comment, marked as answer]
Have a look at the method `$pages->filter()`. This API function is also used by the [blog extension](https://github.com/annaesvensson/yellow-blog) to filter pages in file `blog.php` in function `onParsePageLayout()`. As you can see in the blog source code, it first checks if there are any page request arguments and then filters pages for the blog start page. You can use the same API functions in your custom layout file.

**Comment by PetersOtto** [comment]
Perfect! Thanks for the tip! I got it.
I have added the code to the post https://github.com/datenstrom/yellow/discussions/849#discussioncomment-7726671.

Do you generally see a problem with working with two blogs, as I described above https://github.com/datenstrom/yellow/discussions/933#discussioncomment-8264672? Could that be a problem for Yellow? Maybe performance?

**Reply by markseuffert** [reply]
No, I don't see a problem with either solution. A single layout file is fine. Take the solution which works best for you. 😀

Filtering in the API is fast. Performance testing source code is a great idea and there's something built in to make this super easy. You can activate the debug mode in your system settings with `CoreDebugMode: 1`. Then you should see something like `YellowCore::request status:200 time:19 ms` on the screen. Usually Markdown parsing costs the most CPU time, searching through all pages can take noticeable CPU time, anything else increases/decreases the CPU time probably by a few milliseconds.

**Reply by PetersOtto** [reply]
The fact, that I can also test the performance in debug mode, was not even in my mind. &#128580;
Thx &#128522;
